docker-compose.yml

```
@@ -2,114 +2,350 @@ version: "3"
 services:
   db:
     container_name: db
-    build: ./db
-    restart: always
+    build: ./docker/db
+    restart: unless-stopped
+    healthcheck:
+      test: ["CMD-SHELL", "PGPASSWORD=${POSTGRES_PASSWORD} psql -U ${POSTGRES_USER} ${POSTGRES_DB} -c 'SELECT 1 FROM users LIMIT 1'"]
+      interval: 10s
+      timeout: 10s
+      retries: 10
+      start_period: 1m
     expose:
       - "5432"
     ports:
       - "5431:5432"
     env_file:
-      - ./.env.sample
+      - .env.sndev
     volumes:
+      - ./docker/db/seed.sql:/docker-entrypoint-initdb.d/seed.sql
       - db:/var/lib/postgresql/data
   app:
     container_name: app
-    build: ./
+    build:
+      context: ./
+      args:
+        - UID=${CURRENT_UID}
+        - GID=${CURRENT_GID}
+    restart: unless-stopped
     healthcheck:
       test: ["CMD", "curl", "-f", "http://localhost:3000"]
       interval: 10s
       timeout: 10s
       retries: 10
-      start_period: 1m30s
+      start_period: 1m
     depends_on:
-      - db
+      db:
+        condition: service_healthy
+        restart: true
+      opensearch:
+        condition: service_healthy
+        restart: true
+      sn_lnd:
+        condition: service_healthy
+        restart: true
     env_file:
-      - ./.env.sample
+      - .env.sndev
+    expose:
+      - "3000"
     ports:
       - "3000:3000"
     volumes:
       - ./:/app
     links:
       - db
       - opensearch
+      - sn_lnd
   worker:
     container_name: worker
-    build: ./worker
+    build:
+      context: ./worker
+      args:
+        - UID=${CURRENT_UID}
+        - GID=${CURRENT_GID}
+    restart: unless-stopped
     depends_on:
       db:
-        condition: service_started
+        condition: service_healthy
+        restart: true
       app:
         condition: service_healthy
+        restart: true
+      opensearch:
+        condition: service_healthy
+        restart: true
+      sn_lnd:
+        condition: service_healthy
+        restart: true
     env_file:
-      - ./.env.sample
-    ports:
-      - "8080:8080"
+      - .env.sndev
     volumes:
       - ./:/app
     links:
       - db
       - app
       - opensearch
+      - sn_lnd
     entrypoint: ["/bin/sh", "-c"]
     command:
-      - npm run worker
+      - npm run worker:dev
   imgproxy:
     container_name: imgproxy
     image: darthsim/imgproxy:v3.18.1
     healthcheck:
       test: [ "CMD", "imgproxy", "health" ]
-      timeout: 10s
       interval: 10s
-      retries: 3
-    restart: always
+      timeout: 10s
+      retries: 10
+      start_period: 1m
+    restart: unless-stopped
     env_file:
-      - ./.env.sample
-    expose:
-      - "8080"
+      - .env.sndev
     ports:
       - "3001:8080"
     links:
       - app
   opensearch:
     image: opensearchproject/opensearch:latest
```

We should pin images so everyone uses the same image and it does not depend on when they first ran `docker-compose up`

```suggestion
    image: opensearchproject/opensearch:2.12.0
```

I see below that you already noticed that 2.12.0 requires a strong password set now